app/routes.py

```diff
@@ -1 +1,137 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, make_response, request, abort
+import requests
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+
+def validate_item(cls, request_body):
+    try:
+        new_task = cls.from_dict(request_body)
+    except KeyError:
+        return abort(make_response({"details": "Invalid data"}, 400))
+    return new_task
+
+def validate_id(cls, id):
+    try:
+        id = int(id)
+    except:
+        return abort(make_response({"message": f"{cls.__name__} {id} invalid"}, 400))
+    record = cls.query.get_or_404(id)
+    return record
+
+task_bp = Blueprint("task", __name__, url_prefix="/tasks")
+@task_bp.route("", methods=["POST"])
+def create_one_task():
+    request_body = request.get_json()
+    new_task = validate_item(Task, request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+    return make_response({"task": new_task.to_dict()}, 201)
+
+@task_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+
+    if sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    elif sort_query == "asc":
+        tasks = Task.query.order_by(Task.title)
+    else:
+        tasks = Task.query.all()
+    
+    response = [task.to_dict() for task in tasks]
+    return jsonify(response), 200
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_id(Task, task_id)
+    return make_response({"task": task.to_dict()}, 200)
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = validate_id(Task, task_id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+    return make_response({"task": task.to_dict()}, 200)
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = validate_id(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+    return make_response({"details": f"Task {task.id} \"{task.title}\" successfully deleted"}, 200)
+
+@task_bp.route("/<task_id>/<complete_status>", methods=["PATCH"])
+def mark_complete(task_id, complete_status):
+    task = validate_id(Task, task_id)
+    if complete_status == "mark_complete":
+        task.completed_at = datetime.now()
+        requests.post("https://slack.com/api/chat.postMessage", json={"channel": "task-notifications", "text": f"Someone just completed the task {task.title}"}, headers={"Authorization": os.environ.get("SLACK_BOT_TOKEN")})
```

Comment:
Consider breaking up this line of code by using helper variables. It is not good practice to have lines of code this long.